auditlog/models/rule.py

```diff
@@ -215,41 +215,30 @@ def create(self, vals, **kwargs):
 
     def _make_read(self):
         """Instanciate a read method that log its calls."""
-        # FIXME: read() seems a bit tricky, improve to handle old/new api
-
-        # @api.v7
-        # def read(self, cr, user, ids, fields=None, context=None,
-        #          load='_classic_read', **kwargs):
-        #     print "LOG READ", fields, load, kwargs
-        #     # avoid loops
-        #     if self.env.context.get('auditlog_method_intercepted'):
-        #         return read.origin(
-        #             self, cr, user, ids, fields, context, load, **kwargs)
-        #     # call original method with a modified context
-        #     context = dict(
-        #         self.env.context, auditlog_method_intercepted=True)
-        #     result = read.origin(
-        #         self.with_context(context),
-        #         cr, user, ids, fields, context, load, **kwargs)
-        #     print "RESULT", result
-        #     return result
-
-        # @api.v8
-        # def read(self, fields=None, load='_classic_read', **kwargs):
-        #     print "LOG READ", fields, load, kwargs
-        #     # avoid loops
-        #     if self.env.context.get('auditlog_method_intercepted'):
-        #         return read.origin(self, fields, load, **kwargs)
-        #     # call original method with a modified context
-        #     context = dict(
-        #         self.env.context, auditlog_method_intercepted=True)
-        #     result = read.origin(
-        #         self.with_context(context), fields, load, **kwargs)
-        #     print "RESULT", result
-        #     return result
 
         def read(self, *args, **kwargs):
             result = read.origin(self, *args, **kwargs)
+            # Sometimes the result is not a list but a dictionary
+            try:
```

Comment:
Isn't better to test if it's a list, and if not, encapsulate it in one instead of try/except?